_posts/2017-05-10-polyglot-scientific-computing-with-xtensor.md

```diff
@@ -0,0 +1,239 @@
+---
+layout: post
+title: "Scientific computing in a polyglot world with xtensor"
+date: 2017-05-10 20:20:40 +0100
+author: QuantStack
+categories: [C++]
+excerpt_separator: <!--more-->
+---
+
+![robot](/assets/images/posts/robot.jpg)
+
+<div class="img-caption">
+In the future, there will be actual wars about which language humans should use to speak to the machine.
+</div>
+
+The *One Language to Rule Them All* probably does not exist yet. In the world of interactive scientific computing, countless articles and blogs argue on the merits of R, Python, Julia and other technologies. The truth is, each language has its own ecosystem and qualities. More importantly, these communities are siloed, collaboration is sparse and efforts are duplicated, which hurts sustainability in the long run.
+
+For numerical code, C-level efficiency can be achieved in different ways. Preferred approaches differ in Python and Julia. Library authors for Python often make use of tools such as cython, or try to write code in a way that is amenable to vectorization with numpy. Julia fully relies on just-in-time compilation and the LLVM stack. These two approaches are not easily reconcilable.
+
+As library authors, we directly work with C++ for the numerical code, and provide bindings fo the target languages. This is a usecase in which the `xtensor` C++ library particularly shines.
+
+`xtensor` is a C++ lazy tensor algebra library providing an API similar to that of numpy. You can check out the [numpy to xtensor cheat sheet](http://xtensor.readthedocs.io/en/latest/numpy.html). More importantly, `xtensor` is an *expression system*, which enables the use of tensor expression backed by arbitrary backends such as
```

Comment:
1st->2nd sentence doesn't flow. Maybe sth like: '...API similar to that of numpy, as demonstrated by our [numpy to xtensor cheat sheet](http://xtensor.readthedocs.io/en/latest/numpy.html).'